ui_framework/doc_site/src/views/table/controlled_table.js

```diff
@@ -0,0 +1,214 @@
+import React from 'react';
+
+import {
+  KuiToolBar,
+  KuiToolBarSearchBox,
+  KuiToolBarSection,
+  KuiButton,
+  KuiButtonIcon,
+  KuiTable,
+  KuiToolBarText,
+  KuiControlledTable,
+  KuiTableRowCellLiner,
+  KuiPager,
+  KuiTableRow,
+  KuiTableRowCell,
+  KuiTableRowCheckBoxCell,
+  KuiTableHeaderCell,
+  KuiToolBarFooterSection,
+  KuiToolBarFooter,
+  KuiTableHeaderCheckBoxCell,
+  KuiTableBody,
+  KuiTableHeader,
+} from '../../../../components';
+
+import {
+  LEFT_ALIGNMENT,
+  RIGHT_ALIGNMENT
+} from '../../../../services';
+
+// Arbitrarily use the 3rd index in the sample data as the unique identifier for handling checkbox toggling.
+const CELL_ID_INDEX = 3;
+
+export class ControlledTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedItems: []
```

Comment:
What would you say to drastically simplifying the selection logic? For example, just storing a Map of whether a row is selected or not:

```jsx
this.state = {
  rowToSelectedStateMap: new Map(),
};

toggleItem = (item) => {
  const rowToSelectedStateMap = new Map(this.state.rowToSelectedStateMap);
  rowToSelectedStateMap.set(item, !rowToSelectedStateMap.get(item));
  this.setState({
    rowToSelectedStateMap,
  });
};
```

Then I was thinking that later on we could create a "TableSandbox" page which implements best practices for searching, pagination, sorting, selection, lazy-loading, etc. That would declutter this view. I also think that the current selection logic here is a bit of a strawman, because we would probably construct our rows using a pure data structure, with an ID per row (which we could use for tracking selected state), instead of using arrays of nodes.